src/index.js

```diff
@@ -1,13 +1,136 @@
+  for (let idx = 0; idx < todoListData.length; idx + 1) {
+    todoListData[idx].index = idx;
   }
-  
-  document.body.appendChild(component());
\ No newline at end of file
+};
+
+const saveToLocalStorage = () => {
+  localStorage.setItem('todo_list', JSON.stringify(todoListData));
+};
+
+const addToDo = (input) => {
+  const dataObj = {
+    index: 0,
+    description: '',
+    completed: false,
+  };
+
+  dataObj.description = input;
+  todoListData.push(dataObj);
+  createIndexes();
+  saveToLocalStorage();
+  window.location.reload();
+};
+
+const component = () => {
+  const todoContainer = document.querySelector('.todo-list-container');
+  let element = document.createElement('li');
+  element.className = 'todo-item';
+
+  // Heading
+  const heading = document.createElement('h2');
+  heading.className = 'heading';
+  heading.textContent = 'Today\'s To Do';
+  element.appendChild(heading);
+
+  const clear = document.createElement('button');
+  clear.className = 'clear';
+  clear.innerHTML = '<i class=\'sync alternate icon\'></i>';
+  element.appendChild(clear);
+  todoContainer.appendChild(element);
+
+  // Add todo item
+  element = document.createElement('li');
+  element.className = 'todo-item';
+
+  const addItem = document.createElement('input');
+  addItem.className = 'add-item';
+  addItem.placeholder = 'Add to your list';
+  addItem.value = '';
+  element.appendChild(addItem);
+
+  const enterButton = document.createElement('button');
+  enterButton.className = 'enter-button';
+  enterButton.innerHTML = '<i class=\'level down alternate icon\'></i>';
+  element.appendChild(enterButton);
+  todoContainer.appendChild(element);
+
+  addItem.addEventListener('keydown', (e) => {
+    if (e.keyCode === 13) {
+      addToDo(addItem.value);
+    }
+  });
+
+  enterButton.addEventListener('click', () => {
+    addToDo(addItem.value);
+  });
+
+  // Populate todo items
+  todoListData.forEach((todo) => {
+    element = document.createElement('li');
+    element.className = 'todo-item';
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.className = 'checkbox';
+    checkbox.checked = todo.completed;
+    element.appendChild(checkbox);
+
+    const description = document.createElement('textarea');
+    description.className = 'description';
+    description.rows = 'auto';
+    description.value = todo.description.toLowerCase().charAt(0).toUpperCase();
+    description.value += todo.description.slice(1);
+    element.appendChild(description);
+
+    const taskButton = document.createElement('button');
+    taskButton.className = 'task-button';
+    taskButton.innerHTML = '<i class=\'ellipsis vertical icon\'></i>';
+    element.appendChild(taskButton);
+
+    // description.addEventListener('focus', (e) => {
+    //   e.target.parentNode.style.background = '#ffff0080';
+    //   e.target.parentNode.lastElementChild.innerHTML = `<i class="trash alternate icon"></i>`;
+    // });
```

Comment:
- [ ] OPTIONAL
There are comments found in your code on the repository. Kindly endeavor to remove **all** comments from your code before pushing it to the repository so as to keep your code cleaner.